packages/block-library/src/image/view.js

```diff
@@ -77,269 +77,278 @@ function handleScroll( ctx ) {
+				return ctx.initialized
+					? ctx.imageUploadedSrc
+					: 'data:image/gif;base64,R0lGODlhAQABAAD/ACwAAAAAAQABAAACADs=';
+			},
 		},
-		handleKeydown( event ) {
-			const ctx = getContext();
-			if ( ctx.lightboxEnabled ) {
-				if ( event.key === 'Tab' || event.keyCode === 9 ) {
-					// If shift + tab it change the direction
-					if (
-						event.shiftKey &&
-						window.document.activeElement ===
-							ctx.firstFocusableElement
-					) {
-						event.preventDefault();
-						ctx.lastFocusableElement.focus();
-					} else if (
-						! event.shiftKey &&
-						window.document.activeElement ===
-							ctx.lastFocusableElement
-					) {
-						event.preventDefault();
-						ctx.firstFocusableElement.focus();
-					}
+		actions: {
+			showLightbox( event ) {
+				const ctx = getContext();
+				// We can't initialize the lightbox until the reference
+				// image is loaded, otherwise the UX is broken.
+				if ( ! ctx.imageLoaded ) {
+					return;
 				}
+				ctx.initialized = true;
+				ctx.lastFocusedElement = window.document.activeElement;
+				ctx.scrollDelta = 0;
+				ctx.pointerType = event.pointerType;
+
+				ctx.lightboxEnabled = true;
+				setStyles( ctx, ctx.imageRef );
+
+				ctx.scrollTopReset =
+					window.pageYOffset || document.documentElement.scrollTop;
+
+				// In most cases, this value will be 0, but this is included
+				// in case a user has created a page with horizontal scrolling.
+				ctx.scrollLeftReset =
+					window.pageXOffset || document.documentElement.scrollLeft;
+
+				// We define and bind the scroll callback here so
+				// that we can pass the context and as an argument.
+				// We may be able to change this in the future if we
+				// define the scroll callback in the store instead, but
+				// this approach seems to tbe clearest for now.
+				scrollCallback = handleScroll.bind( null, ctx );
+
+				// We need to add a scroll event listener to the window
+				// here because we are unable to otherwise access it via
+				// the Interactivity API directives. If we add a native way
+				// to access the window, we can remove this.
+				window.addEventListener( 'scroll', scrollCallback, false );
+			},
+			hideLightbox() {
+				const ctx = getContext();
+				ctx.hideAnimationEnabled = true;
+				if ( ctx.lightboxEnabled ) {
+					// We want to wait until the close animation is completed
+					// before allowing a user to scroll again. The duration of this
+					// animation is defined in the styles.scss and depends on if the
+					// animation is 'zoom' or 'fade', but in any case we should wait
+					// a few milliseconds longer than the duration, otherwise a user
+					// may scroll too soon and cause the animation to look sloppy.
+					setTimeout( function () {
+						window.removeEventListener( 'scroll', scrollCallback );
+						// If we don't delay before changing the focus,
+						// the focus ring will appear on Firefox before
+						// the image has finished animating, which looks broken.
```

Comment:
```suggestion
						/*
						 * Delay before changing the focus, otherwise,
						 * the focus ring will appear on Firefox before
						 * the image has finished animating, which looks broken.
						 */
```